src/AutoRest.CSharp/Mgmt/Output/MgmtRestClient.cs

```diff
@@ -37,5 +41,28 @@ public MgmtRestClient(OperationGroup operationGroup, BuildContext<MgmtOutputLibr
             };
             return f;
         }
+
+        public Resource? Resource => _resource ??= GetResource();
+
+        private Resource? GetResource()
+        {
+            if (_evaluatedResource)
+                return _resource;
+
+            Dictionary<string, Resource> candidates = new Dictionary<string, Resource>();
+            _evaluatedResource = true;
+            foreach (var operation in OperationGroup.Operations)
+            {
+                var resource = MgmtExtensions.GetResourceFromResourceType(operation, _context);
+                if (resource is not null)
+                {
+                    candidates.TryAdd(resource.ResourceName, resource);
```

Comment:
What if `TryAdd` fails?

Reply:
Try add won't "fail" it will skip the add if the item already exists which is what we want here